Build Error: Property 'port' does not exist on type 'string | AddressInfo'

#### Do you want to request a _feature_ or report a _bug_?
Bug

#### What's the current behavior?
When building there is a compilation error:
```
src/demos/demoMultiCL.ts:23:63 - error TS2339: Property 'port' does not exist on type 'string | AddressInfo'.
  Property 'port' does not exist on type 'string'.

console.log(`Server listening to ${listener.address().port}`)
```

#### What's the expected behavior?
Build should complete without errors

#### Please tell us about your environment:
N/A

I believe the issue is that @types/node version is not specified in package.json.  It is resolved in package-lock.json but only based on the current installed version of node as it's dependency of other node libraries.

Solution 1: 
Add @types/node to package.json in hopes that it locks it within package-lock.json instead of using `"*"`

Solution 2:
Avoid using `address().port` and simply use outer variable `config.botPort` which was given to server.

--- a/package.json
+++ b/package.json
@@ -44,6 +44,7 @@
     "@types/dotenv": "4.0.3",
     "@types/execa": "^0.8.1",
     "@types/express": "4.16.0",
+    "@types/node": "10.5.3",
     "@types/request-promise": "^4.1.41",
     "@types/jest": "22.2.3",
     "@types/supertest": "2.0.4",

--- a/src/demos/demoMultiCL.ts
+++ b/src/demos/demoMultiCL.ts
@@ -19,8 +19,8 @@ if (isDevelopment) {
     server.use(getDolRouter(config.botPort))
 }
 
-const listener = server.listen(config.botPort, () => {
-    console.log(`Server listening to port: ${listener.address().port}`)
+server.listen(config.botPort, () => {
+    console.log(`Server listening to port: ${config.botPort}`)
 })
 
 const { bfAppId, bfAppPassword, modelId, ...clOptions } = config

--- a/src/demos/demoPasswordReset.ts
+++ b/src/demos/demoPasswordReset.ts
@@ -19,8 +19,8 @@ if (isDevelopment) {
     server.use(getDolRouter(config.botPort))
 }
 
-const listener = server.listen(config.botPort, () => {
-    console.log(`Server listening to port: ${listener.address().port}`)
+server.listen(config.botPort, () => {
+    console.log(`Server listening to port: ${config.botPort}`)
 })
 
 const { bfAppId, bfAppPassword, modelId, ...clOptions } = config

--- a/src/demos/demoPizzaOrder.ts
+++ b/src/demos/demoPizzaOrder.ts
@@ -19,8 +19,8 @@ if (isDevelopment) {
     server.use(getDolRouter(config.botPort))
 }
 
-const listener = server.listen(config.botPort, () => {
-    console.log(`Server listening to port: ${listener.address().port}`)
+server.listen(config.botPort, () => {
+    console.log(`Server listening to port: ${config.botPort}`)
 })
 
 const { bfAppId, bfAppPassword, modelId, ...clOptions } = config

--- a/src/demos/demoStorage.ts
+++ b/src/demos/demoStorage.ts
@@ -18,8 +18,8 @@ if (isDevelopment) {
     server.use(getDolRouter(config.botPort))
 }
 
-const listener = server.listen(config.botPort, () => {
-    console.log(`Server listening to port: ${listener.address().port}`)
+server.listen(config.botPort, () => {
+    console.log(`Server listening to port: ${config.botPort}`)
 })
 
 const { bfAppId, bfAppPassword, modelId, ...clOptions } = config

--- a/src/demos/demoVRAppLauncher.ts
+++ b/src/demos/demoVRAppLauncher.ts
@@ -19,8 +19,8 @@ if (isDevelopment) {
     server.use(getDolRouter(config.botPort))
 }
 
-const listener = server.listen(config.botPort, () => {
-    console.log(`Server listening to port: ${listener.address().port}`)
+server.listen(config.botPort, () => {
+    console.log(`Server listening to port: ${config.botPort}`)
 })
 
 const { bfAppId, bfAppPassword, modelId, ...clOptions } = config

--- a/src/demos/tutorialAPICalls.ts
+++ b/src/demos/tutorialAPICalls.ts
@@ -21,8 +21,8 @@ if (isDevelopment) {
     server.use(getDolRouter(config.botPort))
 }
 
-const listener = server.listen(config.botPort, () => {
-    console.log(`Server listening to port: ${listener.address().port}`)
+server.listen(config.botPort, () => {
+    console.log(`Server listening to port: ${config.botPort}`)
 })
 
 const { bfAppId, bfAppPassword, modelId, ...clOptions } = config

--- a/src/demos/tutorialEntityDetectionCallback.ts
+++ b/src/demos/tutorialEntityDetectionCallback.ts
@@ -19,8 +19,8 @@ if (isDevelopment) {
     server.use(getDolRouter(config.botPort))
 }
 
-const listener = server.listen(config.botPort, () => {
-    console.log(`Server listening to port: ${listener.address().port}`)
+server.listen(config.botPort, () => {
+    console.log(`Server listening to port: ${config.botPort}`)
 })
 
 const { bfAppId, bfAppPassword, modelId, ...clOptions } = config

--- a/src/demos/tutorialHybrid.ts
+++ b/src/demos/tutorialHybrid.ts
@@ -20,8 +20,8 @@ if (isDevelopment) {
     server.use(getDolRouter(config.botPort))
 }
 
-const listener = server.listen(config.botPort, () => {
-    console.log(`Server listening to port: ${listener.address().port}`)
+server.listen(config.botPort, () => {
+    console.log(`Server listening to port: ${config.botPort}`)
 })
 
 const { bfAppId, bfAppPassword, modelId, ...clOptions } = config

--- a/src/demos/tutorialSessionCallbacks.ts
+++ b/src/demos/tutorialSessionCallbacks.ts
@@ -20,8 +20,8 @@ if (isDevelopment) {
     server.use(getDolRouter(config.botPort))
 }
 
-const listener = server.listen(config.botPort, () => {
-    console.log(`Server listening to port: ${listener.address().port}`)
+server.listen(config.botPort, () => {
+    console.log(`Server listening to port: ${config.botPort}`)
 })
 
 const { bfAppId, bfAppPassword, modelId, ...clOptions } = config

--- a/src/server.ts
+++ b/src/server.ts
@@ -11,7 +11,7 @@ if (isDevelopment) {
     app.use(getDolRouter(config.botPort))
 }
 
-const listener = app.listen(config.botPort, () => {
-    console.log(`Server listening to port: ${listener.address().port}`)
+app.listen(config.botPort, () => {
+    console.log(`Server listening to port: ${config.botPort}`)
 })
 

